lms/djangoapps/grades/models.py

```diff
@@ -373,95 +361,50 @@ def update_or_create_grade(cls, **params):
-        try:
-            override = PersistentSubsectionGradeOverride.objects.get(
-                grade__user_id=user_id,
-                grade__course_id=usage_key.course_key,
-                grade__usage_key=usage_key,
-            )
-            if override.earned_all_override is not None:
-                params['earned_all'] = override.earned_all_override
-            if override.possible_all_override is not None:
-                params['possible_all'] = override.possible_all_override
-            if override.earned_graded_override is not None:
-                params['earned_graded'] = override.earned_graded_override
-            if override.possible_graded_override is not None:
-                params['possible_graded'] = override.possible_graded_override
-        except PersistentSubsectionGradeOverride.DoesNotExist:
-            pass
-
         grade, _ = cls.objects.update_or_create(
             user_id=user_id,
             course_id=usage_key.course_key,
             usage_key=usage_key,
             defaults=params,
         )
         if first_attempted is not None and grade.first_attempted is None:
-            if waffle.waffle().is_enabled(waffle.ESTIMATE_FIRST_ATTEMPTED):
-                grade.first_attempted = first_attempted
-            else:
-                grade.first_attempted = now()
+            grade.first_attempted = first_attempted
             grade.save()
 
         cls._emit_grade_calculated_event(grade)
         return grade
 
     @classmethod
-    def _prepare_first_attempted_for_create(cls, params):
-        """
-        Update the value of 'first_attempted' to now() if we aren't
-        using score-based estimates.
-        """
-        if params['first_attempted'] is not None and not waffle.waffle().is_enabled(waffle.ESTIMATE_FIRST_ATTEMPTED):
-            params['first_attempted'] = now()
-
-    @classmethod
-    def create_grade(cls, **params):
-        """
-        Wrapper for objects.create.
-        """
-        cls._prepare_params_and_visible_blocks(params)
-        cls._prepare_first_attempted_for_create(params)
-
-        grade = cls.objects.create(**params)
-        cls._emit_grade_calculated_event(grade)
-        return grade
-
-    @classmethod
-    def bulk_create_grades(cls, grade_params_iter, course_key):
+    def bulk_create_grades(cls, grade_params_iter, user_id, course_key):
         """
         Bulk creation of grades.
         """
         if not grade_params_iter:
             return
 
+        PersistentSubsectionGradeOverride.prefetch(user_id, course_key)
+
         map(cls._prepare_params, grade_params_iter)
         VisibleBlocks.bulk_get_or_create([params['visible_blocks'] for params in grade_params_iter], course_key)
         map(cls._prepare_params_visible_blocks_id, grade_params_iter)
-        map(cls._prepare_first_attempted_for_create, grade_params_iter)
+        map(cls._prepare_params_override, grade_params_iter)
```

Comment:
**Note to reviewers:** Although we may not utilize this unless we delete all grades and recompute them, added a call to `_prepare_params_override` just in case there are any overrides that still need to be applied at the time of re-creating the grades.